File: lab2/Vector.java
import java.util.Arrays;

public class Vector {

    // размер матрицы
    private static Integer size = Lab2.N;

    // генерируем вектор, заполненый случайными числами
    public double[] Random_Vector() {
        double[] vec = new double[size];
        for (int i = 0; i < size; i++) {
            vec[i] = Math.random() * 20.0;
        }
        return vec;
    }

    // заполняем вектора заданым числом
    public double[] Fill_Vector(double number) {
        double[] ResultVector = new double[size];
        for (int i = 0; i < size; i++) {
            ResultVector[i] = number;
        }
        return ResultVector;
    }

    // сумимруем вектора
    public double[] Add_Vec(double[] Vec1, double[] Vec2) {
        for (int i = 0; i < size; i++) {
                Vec1[i] = Vec1[i] + Vec2[i];
        }
        return Vec1;
    }

    // умножение вектора на матрицу
    public double[] Mull_VecxMatr(double[] Vect, double[][] Matr){
        
        double[] ResVector = new double[size];

        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                ResVector[i] = ResVector[i] + Matr[i][j] * Vect[i];
            }
        }
        return ResVector;
    }

    //вывод вектора
    public void Print_Vector(double[] vec) {
        System.out.println("Result: " + Arrays.toString(vec));
    }
}

File: lab6/vector.c
/*********************************************/
/*                                           */
/*                                           */
/*          Lavrinenko Nikita IO-62          */
/*                                           */
/*                                           */
/*********************************************/

#include "opp.h"

float   *Fill_Vector(float num)
{
    float   *vector;

    if(!(vector = (float *)malloc(sizeof(float) * (N + 1))))
        return (NULL);
    for (int i = 0; i < N; i++)
        vector[i] = num;
    return (vector);
}

float   *Add_Vec(float *V1, float *V2)
{
    for (int i = 0; i < N; i++)
        V1[i] += V2[i];
    return (V1);
}

float   *Mull_VecxMatr(float *Vect, float **Matr)
{
    float   *vector;

    if(!(vector = (float *)malloc(sizeof(float) * (N + 1))))
        return (NULL);
    for (int i = 0; i < N; i++)
        for (int j = 0; j < N; j++)
            vector[i] = vector[i] + Matr[i][j] * Vect[i];
    return (vector);
}

void    Print_Vec(float *Vect)
{
    printf("| ");
    for (int i = 0; i < N; i++)
        printf("%f ", Vect[i]);
    printf("|\n");
}